functions/controllers/user.js

```diff
@@ -0,0 +1,135 @@
+const { status, message } = require("../util/constants");
+const config = require("../util/config");
+const firebase = require("firebase");
+var HttpStatus = require("http-status-codes");
+require("firebase/firestore");
+firebase.initializeApp(config);
+
+const db = firebase.firestore();
+
+const { validateSignUpData, validateLoginData } = require("../util/validator");
+
+exports.signupUser = async (req, res) => {
+  const {
+    fullName,
+    userName,
+    email,
+    number,
+    password,
+    confirmPassword,
+    userStatus
+  } = req.body;
+  const newUser = {
+    fullName,
+    userName,
+    email,
+    number,
+    password,
+    confirmPassword,
+    userStatus
+  };
+  const { error, success } = status;
+  const {
+    somethingWentWrong,
+    wrongCredentials,
+    auth_emailInUse,
+    auth_userNotFound,
+    auth_wrongPassword,
+    emailInUse,
+    userNameExists
+  } = message;
+  const {
+    PRECONDITION_FAILED,
+    INTERNAL_SERVER_ERROR,
+    BAD_REQUEST,
```

Comment:
cleanup this file
create a User folder under controller and have a
index.js ->convert this current file to index.js and call signup and login part there
signup.js ->extract line 12-98 into signup.js
login.js ->extract the login part into login.js